src/nfa/operations.cc

```diff
@@ -664,11 +690,13 @@ bool Mata::Nfa::is_lang_empty(const Nfa& aut, Run* cex)
 } // is_lang_empty }}}
 
 
+//method?
 Nfa Mata::Nfa::Algorithms::minimize_brzozowski(const Nfa& aut) {
     //compute the minimal deterministic automaton, Brzozovski algorithm
     return determinize(revert(determinize(revert(aut))));
 }
 
+//method?
```

Comment:
We agreed that functions should return new instance and methods should compute inplace, if at all possible. Therefore, this function should remain a function.

Reply:
In that case, we could use chaining and write something like `intersection(nfa1.trim().revert(), nfa2.determinize())`. This would allow us to eliminate some optional parameters from several functions, I believe.

Reply:
I am not sure if the in-place implementation brings us something here (performance wise). I don't know how to do quotienting efficiently on our structure of NFA without a fresh new construction.

Reply:
hm, true, one could avoid some of the allocations, not all.   

The main question is whether we really want this dogma : "not in place therefore function" and not use methods that return new automata.
Maybe I just don't remember what the arguments were. 
I

Reply:
It is more a matter of taste. It would be consistent if we kept the dogma, but there is not much to gain from that. I would be OK with using both approaches interchangeably. For example, we can have one (or more) from these options implemented:
```cpp
Nfa Mata::Nfa::concatenate(const Nfa& nfa1, const Nfa& nfa2);
Nfa Mata::Nfa::concatenatenation(const Nfa& nfa1, const Nfa& nfa2);
Nfa Mata::Nfa::concatenated(const Nfa& nfa1, const Nfa& nfa2);
Nfa Mata::Nfa::Nfa::concatenated_with(const Nfa& nfa2);
Nfa Mata::Nfa::Nfa::concatenated_with(const Nfa& nfa2);
```
All the options are acceptable, in my opinion. But maybe for operations working with two automata, having functions is a better design, clearly showing that the original automata are not modified.